### ISSUE TITLE
Trailing assignments on next line are missing continuation indent in 0.43.0

### ISSUE THREAD

**Issue by ZacSweers** [pinterest/ktlint#1268]
## Expected Behavior

I expect it to be indented

## Observed Behavior

It is indenting to the same indent as the previous line (screenshot is before/after diff when running 0.43.0)

![image](https://user-images.githubusercontent.com/1361086/139927852-39fb0f6e-e459-4b96-af21-57c399b8bc9a.png)

## Steps to Reproduce

```kotlin
private val richTextSections =
  listOf(
    RichTextSection(
      chunks =
        // This gets indented back to align with the chunks name
        listOf(
          TextChunk(text = "test")
        )
    )
  )
```

## Your Environment

* Version of ktlint used: 0.43.0
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): Spotless
* Version of Gradle used (if applicable): N/A
* Operating System and version: N/A
* Link to your project (if it's a public repository): N/A

**Comment by ZacSweers** [comment]
Actually it appears to affect everything in the expression after the assignment, not just the start of it

![image](https://user-images.githubusercontent.com/1361086/139928725-7dca5fc0-f31f-4582-a04a-9ae97b8f089d.png)

**Comment by shashachu** [comment]
@ZacSweers thanks for the report. Are you able to run with the `--verbose` flag to confirm which rule is causing the issue?

**Comment by ZacSweers** [comment]
I don't see any extra output with that flag

```
$ ./config/bin/ktlint -F --verbose /Users/zsweers/path/to/file.kt
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.jetbrains.kotlin.com.intellij.util.ReflectionUtil (file:/Users/zsweers/dev/slack/restructure/slack-android-ng/config/bin/ktlint) to field java.lang.Throwable.backtrace
WARNING: Please consider reporting this to the maintainers of org.jetbrains.kotlin.com.intellij.util.ReflectionUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

```

**Comment by romtsn** [comment]
@ZacSweers please run it w/o the `-F` option, then it should give the affected lines and the rule id

**Comment by ZacSweers** [comment]
Ah ok, here you go

```
$ ./config/bin/ktlint --verbose path/to/file.kt
path/to/file.kt:67:1: Unexpected indentation (10) (should be 8) (indent)
path/to/file.kt:68:1: Unexpected indentation (12) (should be 10) (indent)
path/to/file.kt:69:1: Unexpected indentation (10) (should be 8) (indent)

```

**Comment by paul-dingemans** [comment]
This was explicitly implemented in #964. I have checked, and this is still the way that IntelliJ Ultimate formats the example. Personally, I would prefer to have the indented it as well ;-)

**Comment by DenWav** [comment]
Wait so this is intended behavior? That can't be write, it's clearly wrong.

**Comment by paul-dingemans** [comment]
> This was explicitly implemented in #964. I have checked, and this is still the way that IntelliJ Ultimate formats the example. Personally, I would prefer to have the indented it as well ;-)

Removing the "bug" label and add "won't fix".
